querydb/src/main/scala/io/joern/scanners/android/AndroidMisconfigurations.scala

```diff
@@ -104,4 +104,95 @@ object AndroidMisconfigurations extends QueryBundle {
            |</manifest>""".stripMargin)
       )
     )
+
+  // todo: check for `android:filterTouchesWhenObscured="[true|false]"`
+  // todo: check for `onFilterTouchEventForSecurity`
+  // see: https://redfoxsec.com/blog/android-tapjacking-vulnerability/
+  // see: https://cwe.mitre.org/data/definitions/1021.html
+  // see: https://infinum.com/blog/pentesting-misc-cases/
+  // see: https://developer.android.com/reference/android/Manifest.permission.html#SYSTEM_ALERT_WINDOW
+  @q
+  def tapJacking(): Query =
+    Query.make(
+      name = "tap-jacking",
+      author = Crew.claudiu,
+      title = "Tap Jacking: target SDK <23 specified in `build.gradle` ",
+      description = """
+          |Android apps targeting API levels 22 and lower have the SYSTEM_ALERT_WINDOW permission enabled by default.
+          |This allows apps to draw overlays over other apps. Attackers can use this option to create an overlay that
+          |would essentially hijack user taps and use it to obtain sensitive user information.""".stripMargin,
+      score = 6,
+      withStrRep({ cpg =>
+        def groovyBuildGradleFiles                                  = cpg.configFile.name(".*build.gradle")
+        val targetSdkVersionMatch                                   = """^[^t]+targetSdk[^0-9]+(\d+)""".r
+        val minimumAndroidSdkVersionWhereNoAdditionalChecksRequired = 23
+        groovyBuildGradleFiles.filter { gradleFile =>
+          gradleFile.content
+            .split('\n')
+            .filter { line =>
+              targetSdkVersionMatch
+                .findAllIn(line)
+                .matchData
+                .filter { m =>
+                  m.groupCount > 0 && m.group(1).toInt < minimumAndroidSdkVersionWhereNoAdditionalChecksRequired
+                }
+                .nonEmpty
```

Comment:
my IntelliJ would show some squiggles here (and 1 line below) to simplify `.filter(...).nonEmpty` to `.exists`